apps/hash-agents/.gitignore

```diff
@@ -0,0 +1,3 @@
+.env
```

Comment:
`.env.local` is not a default at all, so generally this leads to more problems than it solves in most cases. I'm not aware of any location in the repository where it's used.

Reply:
Shouldn't the root .gitignore superseed this anyway? perhaps we can remove this

Reply:
The root `.gitignore` does not mention the `.env`, only `.env.local` (which we don't use anymore)

Reply:
> I'm not aware of any location in the repository where it's used.

We are (or at least I am) using `.env.local` files for local development of TS backend modules. That's why we're using the `dotenv-flow` module in the shared `libs/@local/hash-backend-utils/src/environment.ts` file, and why we are git ignoring `.env.local` in the repo root.

We are currently committing `.env` to the repository in the root directory, so if we use `.env` for sensitive un-committed information in this directory it'd be slightly inconsistent behaviour. But you've made me realise we are already doing this in other places, so maybe not worth worrying about for this PR.

Reply:
cc @CiaranMn re: ensuring a consistent (and good DX) `.env` approach

Reply:
I spoke with @thehabbos007 this morning about this and we want to have both, `.env.local` and `.env`. This will probably end up with the best DX

Reply:
I agree with the above suggestion as being the best for now since it's consistent with how we are doing it for HASH (but not blocks).

As to how to do this consistently (a future change), maybe `.env.default` for committed things, and `.env` for secret things (or `.env.local` for secret things). I don't think a file called `.env` that gets committed is the most natural approach.

Reply:
I've added support for `.env.local` in a37ccc6ae908b9ed25de3b39b497e0c9fd392cfc. Please make sure you don't commit your `.env` file with the secret in it as this is not ignored anymore!